app/src/main/java/com/android/sample/utils/NetworkUtils.kt

```diff
@@ -0,0 +1,23 @@
+package com.android.sample.utils
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+
+/** Utility class for network-related operations. */
+open class NetworkUtils {
+
+  /**
+   * Checks if the network is available.
+   *
+   * @param context The context used to get the ConnectivityManager.
+   * @return True if the network is available, false otherwise.
+   */
+  fun isNetworkAvailable(context: Context): Boolean {
+    val connectivityManager =
+        context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+    val network = connectivityManager.activeNetwork ?: return false
+    val networkCapabilities = connectivityManager.getNetworkCapabilities(network) ?: return false
+    return networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
+  }
```

Comment:
That's a great question! Since I currently rely on 4G and don’t have Wi-Fi at home, I didn’t test it under those specific conditions, to be honest. However, I did test it on the emulator with Wi-Fi and Airplane mode enabled, and the result was that `isNetworkAvailable` returned `true`.

Reply:
Yes, I tested it as well, and it worked in the same way. I think this is an effective method to implement it!